### ISSUE TITLE
Unknown flag: chown

### ISSUE THREAD

**Issue by jpswade** [moby/moby#35731]
**Description**

When using `COPY --chown=user ...` you get the following error:

"Unknown flag: chown"

**Steps to reproduce the issue:**
1. Using docker version 17.11.0-ce
2. Have a Dockerfile that contains `COPY --chown=user ...`
3. docker build
4. ???
5. Profit!

**Describe the results you received:**

I got the error: Unknown flag: chown

**Describe the results you expected:**

I expected no error, I expected it to recognise the flag.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
$ docker version
Client:
 Version:      17.11.0-ce
 API version:  1.23
 Go version:   go1.9.2
 Git commit:   1caf76c
 Built:        unknown-buildtime
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.6
 API version:  1.24 (minimum version )
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Wed Jan 11 00:23:16 2017
 OS/Arch:      linux/amd64
 Experimental: false

```

**Output of `docker info`:**

```
$ docker info
Containers: 63
 Running: 29
 Paused: 0
 Stopped: 34
Images: 291
Server Version: 1.12.6
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
 Log: 
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 
runc version: 
init version: 
Security Options:
 seccomp
Kernel Version: 4.9.13
Operating System: Buildroot 2017.02
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.954GiB
Name: minikube
ID: WPRL:VAL2:NMPG:VFQ4:HD5D:Y3QN:H5QE:C2M2:KNUH:X3QY:OLKG:H6XO
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: XXX
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Experimental: false
Insecure Registries:
 10.0.0.0/24
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by cpuguy83** [comment]
That's because your server version is 1.12 which does not support the flag.
Ease make sure to upgrade the server.

Thanks!

**Comment by iahmad-khan** [comment]
Step 2/5 : COPY --chown=elasticsearch:elasticsearch elasticsearch.yml /usr/share/elasticsearch/config/
Unknown flag: chown
[root@wp-p3s-1c elk]# docker version
Client:
 Version:         1.13.1
 API version:     1.26
 Package version: <unknown>
 Go version:      go1.8.3
 Git commit:      632fb37
 Built:           Mon Nov 27 15:57:40 2017
 OS/Arch:         linux/amd64

Server:
 Version:         1.13.1
 API version:     1.26 (minimum version 1.12)
 Package version: <unknown>
 Go version:      go1.8.3
 Git commit:      632fb37
 Built:           Mon Nov 27 15:57:40 2017
 OS/Arch:         linux/amd64
 Experimental:    false

**Comment by cpuguy83** [comment]
1.13 also does not have this flag. I believe it was added in Docker CE 17.06.

**Comment by iahmad-khan** [comment]
ok thanks

**Comment by mhabsaoui** [comment]
Hi,

Even with new versions on d/d-c, i got same error:

> Step 4/5 : COPY --chown=mysql:mysql injected/script.sql /docker-entrypoint-initdb.d/script.sql
ERROR: Service 'mysql' failed to build: Unknown flag: chown

Info:

> docker-compose version 1.18.0, build 8dd22a9
Docker version 17.05.0-ce, build 89658be

Thanks for help

**Comment by jsixface** [comment]
I am using version 17.06 and still not working

```
Step 20/24 : COPY --chown=tomcat:tomcat --from=builder /srv/tomcatServices/webapps/*.war /srv/tomcatServices/webapps/
Unknown flag: chown
arumugam@poc1847-tmp>> docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:17:04 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:15:57 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by thaJeztah** [comment]
It was implemented in https://github.com/moby/moby/pull/34263, which was merged in August, so I think it's 17.09 and up

**Comment by deerawan** [comment]
## Problem
I also got this issue when trying to execute command
```
COPY --chown=node:node
```

My spec
```
OSX macOS Sierra
```
```
$ docker version
Client:
 Version:       17.12.0-ce
 API version:   1.28 (downgraded from 1.35)
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:03:51 2017
 OS/Arch:       darwin/amd64

Server:
 Engine:
  Version:      17.04.0-ce
  API version:  1.28 (minimum version 1.12)
  Go version:   go1.7.5
  Git commit:   4845c56
  Built:        Wed Apr  5 18:45:47 2017
  OS/Arch:      linux/amd64
  Experimental: false
```

The problem is my Server version is `17.04.0-ce` whereas `copy --chown` is supported in > `17.09.x`

## Solution
I solved it by updating my docker server
```
docker-machine upgrade default
```
`default` is my docker machine name
